packages/loader/driver-utils/src/parallelRequests.ts

```diff
@@ -474,13 +486,31 @@ export function requestOps(
             queue.pushValue(deltas);
         });
 
+    // Implement faster cancellation. getSingleOpBatch() checks signal, but only in between
+    // waits (up to 10 seconds) and fetches (can take infinite amount of time).
+    // While every such case should be improved and take into account signal (and thus cancel immediately),
+    // it is beneficial to have catch-all
+    const listener = (event: Event) => { manager.cancel(); };
+    if (signal !== undefined) {
+        signal.addEventListener("abort", listener);
+    }
+
     manager.run(concurrency)
-        .then(() => {
-            telemetryEvent.end({
+        .finally(() => {
+            if (signal !== undefined) {
+                signal.removeEventListener("abort", listener);
+            }
+        }).then(() => {
+            const props = {
                 lastFetch,
                 deltasRetrievedTotal,
                 requests,
-            });
+            };
+            if (manager.canceled) {
+                telemetryEvent.cancel({ ...props });
```

Comment:
We can just pass the props here and not do { ...props }?

Reply:
Yep, left behind after experimenting with adding props, will fix it.